vector_db.py

```
@@ -0,0 +1,77 @@
+    def __init__(self, index_path: str = "faiss_index.bin", metadata_path: str = "faiss_metadata.pkl"):
+        self.index_path = index_path
+        self.metadata_path = metadata_path
+        self.index = None
+        self.metadata = []  # список drawing_id в порядке добавления
+        self._load_or_create()
+
+    def _load_or_create(self):
+        if os.path.exists(self.index_path) and os.path.exists(self.metadata_path):
+            self.index = faiss.read_index(self.index_path)
+            with open(self.metadata_path, "rb") as f:
+                self.metadata = pickle.load(f)
```

Ruff (0.15.12)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)